resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -184,4 +183,18 @@ object YarnSparkHadoopUtil {
     ConverterUtils.toContainerId(containerIdString)
   }
 
+  /**
+   * Convert MEMORY_OFFHEAP_SIZE to MB Unit, return 0 if MEMORY_OFFHEAP_ENABLED is false.
+   */
+  def executorOffHeapMemorySizeAsMb(sparkConf: SparkConf): Int = {
+    if (sparkConf.get(MEMORY_OFFHEAP_ENABLED)) {
+      val sizeInMB =
+        sparkConf.getSizeAsMb(MEMORY_OFFHEAP_SIZE.key, MEMORY_OFFHEAP_SIZE.defaultValueString).toInt
```

Comment:
personally not a fan of this way, can we just do a. conf.get(MEMORY_OFFHEAP_SIZE) and then call the byteStringAsMb.   That way we aren't explicitly having to get key and default value and we are bypassing the ConfigBuilder stuff.

Reply:
[8f37ca1](https://github.com/apache/spark/pull/25309/commits/8f37ca13c2bd803e75151fe5e701c2f1a80d4478) change to use `Utils.byteStringAsMb(sparkConf.get(MEMORY_OFFHEAP_SIZE).toString).toInt` to conversion config value to sizeInMB.

Reply:
@tgravescs ref api comments, 
We have to use `s"${sparkConf.get(MEMORY_OFFHEAP_SIZE)}B"` as input to call  `Utils.byteStringAsMb(str: String)` because `byte string should 50b, 100k, or 250m`,looks a little strange.